spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/PropertyLoader.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.autoconfigure.telemetry;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.java.Log;
+import org.springframework.lang.NonNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+@Log
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class PropertyLoader {
+
+    private static final String PROJECT_PROPERTY_FILE = "/META-INF/project.properties";
+    private static final String APPLICATION_PROPERTY_FILE = "/application.properties";
+    private static final String APPLICATION_YML_FILE = "/application.yml";
+
+    public static String getProjectVersion() {
+        return getPropertyByName("project.version", PROJECT_PROPERTY_FILE);
+    }
+
+    public static boolean isApplicationTelemetryAllowed() {
+        String telemetryAllowed = getPropertyByName("spring.cloud.azure.telemetryAllowed", APPLICATION_PROPERTY_FILE);
+
+        if (telemetryAllowed == null) {
+            telemetryAllowed = getPropertyByName("telemetryAllowed", APPLICATION_YML_FILE);
+        }
+
+        if (telemetryAllowed == null) {
+            return true;
+        } else {
+            return telemetryAllowed.equalsIgnoreCase("false") ? false : true;
+        }
+    }
+
+    private static String getPropertyByName(@NonNull String name, @NonNull String filename) {
+        final Properties properties = new Properties();
+        final InputStream inputStream = PropertyLoader.class.getResourceAsStream(filename);
+
+        if (inputStream == null) {
```

Comment:
Returning null is confusing here, since user couldn't distinguish between property is null or resource file not existed.